herd/libdir/arm-models/mixed/ec.cat

```diff
@@ -40,17 +40,24 @@
 include "armv8-common.cat"
 include "sca.cat"
 
-let preorder-cb = IM0 | lob | scaob
+let preorder-cb = IM0 | local-hw-reqs | scaob
 let preorder-cb-lift = lift(MC,preorder-cb)
 
 with cb from linearisations(MC,preorder-cb-lift)
 ~empty cb
 
 let dcbl = delift(cb) & loc
-let rf-fwd = (W*R) & po-loc & dcbl^-1 \ intervening-write(po-loc)
-let rf-nfwd = (W*R) & dcbl \ intervening-write(dcbl | po-loc)
+let rf-fwd = (W*(Exp&R)) & po-loc & dcbl^-1 & ~intervening(W,po-loc)
+let rf-nfwd = (W*(Exp&R)) & dcbl & ~(intervening(W, (dcbl | po-loc)); [Exp & R]; po-loc?)
```

Comment:
This not equivalent to the previous formulation (due to the suffix `[Exp & R]; po-loc`, I am just marking it to make sure this is intentional.

Reply:
Indeed intentional: `coRR` shapes are now a part of the external visibility requirements, so appropriate compensation is necessary for external completion and external global completion.

Reply:
Updated this by unpacking one clause into multiple -- as it is easier to explain them separately. Marking as resolved.